File: ios/PKBE/SessionStore.swift
import Foundation

enum DeviceIdentity {
    private static let key = "pkbe.deviceId"

    static var deviceId: String {
        if let existing = UserDefaults.standard.string(forKey: key) {
            return existing
        }
        let created = UUID().uuidString.lowercased()
        UserDefaults.standard.set(created, forKey: key)
        return created
    }
}

enum SessionStore {
    private static let key = "pkbe.token"

    static var token: String? {
        get { UserDefaults.standard.string(forKey: key) }
        set { UserDefaults.standard.set(newValue, forKey: key) }
    }

    static var username: String? {
        get { UserDefaults.standard.string(forKey: "pkbe.username") }
        set { UserDefaults.standard.set(newValue, forKey: "pkbe.username") }
    }

    static func clear() {
        token = nil
        username = nil
    }
}

File: android/app/src/main/java/com/uci/pkbe/SessionStore.kt
package com.uci.pkbe

import android.content.Context
import java.util.UUID

object DeviceIdentity {
    private const val PREFS = "pkbe"
    private const val KEY = "deviceId"

    fun deviceId(context: Context): String {
        val prefs = context.getSharedPreferences(PREFS, Context.MODE_PRIVATE)
        val existing = prefs.getString(KEY, null)
        if (!existing.isNullOrBlank()) return existing
        val created = UUID.randomUUID().toString().lowercase()
        prefs.edit().putString(KEY, created).apply()
        return created
    }
}

object SessionStore {
    private const val PREFS = "pkbe"

    fun token(context: Context): String? =
        context.getSharedPreferences(PREFS, Context.MODE_PRIVATE).getString("token", null)

    fun username(context: Context): String? =
        context.getSharedPreferences(PREFS, Context.MODE_PRIVATE).getString("username", null)

    fun save(context: Context, token: String, username: String) {
        context.getSharedPreferences(PREFS, Context.MODE_PRIVATE).edit()
            .putString("token", token)
            .putString("username", username)
            .apply()
    }

    fun clear(context: Context) {
        context.getSharedPreferences(PREFS, Context.MODE_PRIVATE).edit()
            .remove("token")
            .remove("username")
            .apply()
    }
}
